File: app/src/main/java/com/bkjcb/rqapplication/fragment/CheckResultFragment.java
package com.bkjcb.rqapplication.fragment;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bkjcb.rqapplication.CreateCheckTaskActivity;
import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.model.CheckItem;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * Created by DengShuai on 2020/2/21.
 * Description :
 */
public class CheckResultFragment extends BaseSimpleFragment {

    @BindView(R.id.info_type)
    TextView mInfoType;
    @BindView(R.id.info_station)
    TextView mInfoStation;
    @BindView(R.id.info_date)
    TextView mInfoDate;
    @BindView(R.id.info_year)
    TextView mInfoYear;
    @BindView(R.id.info_result)
    TextView mInfoResult;
    @BindView(R.id.info_remark)
    TextView mInfoRemark;
    @BindView(R.id.info_name)
    TextView mInfoName;
    @BindView(R.id.info_edit)
    ImageView mInfoEdit;
    protected CheckItem checkItem;

    public void setCheckItem(CheckItem checkItem) {
        this.checkItem = checkItem;
    }

    public static CheckResultFragment newInstance(CheckItem checkItem) {
        CheckResultFragment fragment = new CheckResultFragment();
        fragment.setCheckItem(checkItem);
        return fragment;
    }

    public void updateData(CheckItem checkItem) {
        this.checkItem = checkItem;
        initData();
    }

    @Override
    public void setResId() {
        this.resId = R.layout.fragment_check_result;
    }

    @Override
    protected void initView() {
        if (checkItem.status > 0) {
            mInfoEdit.setVisibility(View.GONE);
        }
    }

    @OnClick({R.id.info_edit})
    public void OnClick(View view) {
        CreateCheckTaskActivity.ToActivity(getContext(), checkItem);
    }

    @Override
    protected void initData() {
        if (checkItem != null) {
            mInfoType.setText(checkItem.zhandianleixing);
            mInfoStation.setText(checkItem.beijiandanwei);
            mInfoYear.setText(checkItem.year);
            mInfoDate.setText(checkItem.jianchariqi);
            mInfoRemark.setText(checkItem.beizhu);
            mInfoResult.setText(checkItem.jianchajieguo);
            mInfoName.setText(checkItem.checkMan);
        }
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/check/fragment/CheckResultFragment.kt
package com.bkjcb.rqapplication.check.fragment

import android.view.View
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.check.CreateCheckTaskActivity
import com.bkjcb.rqapplication.check.model.CheckItem
import com.bkjcb.rqapplication.base.fragment.BaseSimpleFragment
import kotlinx.android.synthetic.main.fragment_check_result.*

/**
 * Created by DengShuai on 2020/2/21.
 * Description :
 */
open class CheckResultFragment : BaseSimpleFragment() {
    protected var checkItem: CheckItem? = null

    fun updateData(checkItem: CheckItem?) {
        this.checkItem = checkItem
        initData()
    }


    override fun initView() {
        if (checkItem!!.status > 0) {
            info_edit.visibility = View.GONE
        }
        info_edit.setOnClickListener {
            CreateCheckTaskActivity.toActivity(context!!, checkItem)
        }
    }

    override fun initData() {
        if (checkItem != null) {
            info_type.text = checkItem!!.zhandianleixing
            info_station.text = checkItem!!.beijiandanwei
            info_year.text = checkItem!!.year
            info_date.text = checkItem!!.jianchariqi
            info_remark.text = checkItem!!.beizhu
            info_result.text = checkItem!!.jianchajieguo
            info_name.text = checkItem!!.checkMan
        }
    }

    override fun initResID() = R.layout.fragment_check_result

    companion object {
        fun newInstance(checkItem: CheckItem?): CheckResultFragment {
            val fragment = CheckResultFragment()
            fragment.checkItem = checkItem
            return fragment
        }
    }
}